### ISSUE TITLE
Memory Leak heads up

### ISSUE THREAD

**Issue by leobeosab** [clarketm/s3recon#6]
When scanning with a large list ~9000 it will eat all available RAM I have 32GBs and 31 of it was used. I can also make a PR if that helps :)

**Comment by clarketm** [comment]
> When scanning with a large list ~9000 it will eat all available RAM I have 32GBs and 31 of it was used. I can also make a PR if that helps :)

Thanks for the issue. If you have a fix, a PR would be great!

**Comment by leobeosab** [comment]
> Thanks for the issue. If you have a fix, a PR would be great!

Sounds great, I'll create one when I get some more time. It just needs a limit on the number of concurrent tasks. :)

**Comment by clarketm** [comment]
Resolved this in [v1.4.0](https://github.com/clarketm/s3recon/releases/tag/v1.4.0): https://github.com/clarketm/s3recon/commit/d00fdedfe325152137a934f18dafb62d12dbdc60 

Also, # of concurrent requests is configurable via a flag ([docs](https://s3recon.readthedocs.io/en/latest/)):
```console
-c num, --concurrency num      maximum <num> of concurrent requests (default: # of lcpus)
```
